### DISCUSSION TITLE
delete user RPC causes subsequent signOut request to fail

### DISCUSSION THREAD

**Discussion by anngbaum** [supabase/supabase#9327] [Questions]
we need to allow users to delete their own accounts on our platform. we implemented an RPC and it executes fine. we also enforce RLS on the table so that only a logged in user can delete his or her own account.

however, once `deleteUser` is called, we then need to clear the local cache and session. the auth `signOut` request fails with a 500. this creates problems on our web client, where we hold onto the cached session data until the token expires.

the only discussion I saw was from over a year ago, with this issue called out explicitly here: https://github.com/supabase/supabase/discussions/1066#discussioncomment-727319

is there a recommended way for "forcing" the client to sign out? or do we need to manage session ourselves if we want to implement a `deleteUser` RPC?

alternatively, maybe this is a request for the AuthClient to support a "local signout" if the user has been deleted?

**Comment by kangmingtay** [comment]
hi @anngbaum, have you tried calling the `signOut` first to remove the local session for the user and run the `deleteUser` RPC after the signOut is completed to continue to delete the user?

**Reply by silentworks** [reply]
You probably can't call `signOut` first as you wouldn't be able to run the `.rpc` function since there is no JWT to check against to make sure the user is allowed to delete the account.

**Reply by anngbaum** [reply]
^^this is exactly the problem we're encountering – we naturally want deleteUser to be RLS protected so that users can't go deleting other people's accounts, so we need to be logged in when the `rpc` gets executed.

one of our team members identified a solution using `localStorage.removeItem(process.env.PUBLIC_STORAGE_KEY);` and adding that storage key when creating the supabase client. that seems the best option for now, though this might be a nice feature request if a deleteUser function is explored further in the future!

**Comment by anngbaum** [comment]
just in the interest of spring cleaning, this appears to have been fixed in https://github.com/supabase/gotrue-js/pull/541

**Comment by paule89123** [comment]
I don't think this has been fixed.

Calling supabase.auth.signOut() still fails if the user has been deleted.

**Comment by SebasScript** [comment]
Its not working when trying to call sing out after user has been deleted the auth.signOut() throws an error

```
AuthApiError: User from sub claim in JWT does not exist
    at handleError (fetch.ts:71:9)
    at async _handleRequest (fetch.ts:150:5)
    at async _request (fetch.ts:117:16)
    at async GoTrueAdminApi.signOut (GoTrueAdminApi.ts:65:7)
    at async GoTrueClient.ts:1491:27
    at async SupabaseAuthClient._useSession (GoTrueClient.ts:991:14)
    at async SupabaseAuthClient._signOut (GoTrueClient.ts:1484:12)
    at async GoTrueClient.ts:1477:14
```

**Reply by saffiullahkhugyani** [reply]
It was the same error for me as well, but the funniest thing is I was Testing this scenario on a separate project, I developed a Flutter project in which I sign in as a user, log in, sign out, and delete using the RPC function,  and it works like a charm, however, when I added the same functionality only the difference was that I put the code into an architectural design pattern, it gives me the same error "User from sub claim in JWT does not exist when signout ". I am confused
